packages/plugin/webpack/src/Config.ts

```diff
@@ -48,6 +49,20 @@ export interface WebpackPluginEntryPoint {
    * for all entries.
    */
   nodeIntegration?: boolean;
+  /**
+   * Custom options to merge into the configuration passed to `HtmlWebpackPlugin`.
+   */
+  htmlOptions?: Partial<HtmlWebpackPlugin.Options>;
+  /**
+   * Plugins to include before `HtmlWebpackPlugin`; typically, HTML plugin add-ons will
+   * need to be placed here.
+   */
+  htmlPlugins?: WebpackPluginInstance[];
```

Comment:
Instead of passing in a random list of plugins here, could we just change our implementation to ensure the config is extended with user plugins appearing **first** in the list instead of always adding ours before?  That would avoid this extra configuration option completely

Reply:
the intent of isolating this as its own property instead of adjusting the plugin flow is three-fold:

**1) less invasive.**
i suspect downstream projects rely on the order of plugins, so i am doing my best not to adjust the _existing_ execution order in any way that might affect how existing projects work.

**2) no way to include only where HTML targets exist.**
for example, say you have several targets, a few with exotic renderers or something, and then just one target with `HtmlWebpackPlugin`. if you use the global `plugins` property, it will apply to all entries instead of just the applicable invocation of `HtmlWebpackPlugin`.

**3) semantic meaning.**
because these are the only plugins that must run before `HtmlWebpackPlugin`, this should keep the plugin flow easy to reason about for future changes.

these are just the thoughts behind this approach. i am totally open to thoughts on better typing or alternate approaches.